packages/typescript-resolver-files-e2e/src/test-extended-object-types/schema-federated-extended-object-type/topic/resolvers/ForeignType.ts

```diff
@@ -0,0 +1,7 @@
+import type { ForeignTypeResolvers } from './../../types.generated';
+export const ForeignType: Pick<
+  ForeignTypeResolvers,
+  'id' | 'topics' | '__resolveReference'
```

Comment:
Should both `ForeignType` be able to pick `__resolveReference` ? Since we are merging in `resolvers.generated.ts`, one `__resolveReference` would be overridden. 

I wonder if we should only allow the one with `@key` directive to pick `__resolveReference`?

Reply:
I originally set out to do this, but wasn't able to spot a clean way to do that with the current set-up. I'm probably missing something very obvious.

> I wonder if we should only allow the one with @key directive to pick __resolveReference?

I guess this relates to your above question, but I believe it is valid for a GraphQL type to have multiple object directives defined across many `extends`. I haven't been able to find anything concrete in the spec though

Reply:
> I originally set out to do this, but wasn't able to spot a clean way to do that with the current set-up. I'm probably missing something very obvious.

Ah yes, you are right! I remember now that the parsed GraphQL schema AST doesn't give this information. That's something I can take a look later. There's a small risk of overriding `__resolveReference`  but that can be patched later.

Thanks for looking into all the options!